streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -791,12 +788,16 @@ public boolean conditionMet() {
 
         final ReadOnlyKeyValueStore<String, String> store2 = kafkaStreams.store(storeName, QueryableStoreTypes.<String, String>keyValueStore());
 
-        TestUtils.waitForCondition(new TestCondition() {
-            @Override
-            public boolean conditionMet() {
-                return "125".equals(store2.get("a")) && "34".equals(store2.get("b"));
-            }
-        }, maxWaitMs, "wait for agg to be '123'");
+        try {
+            TestUtils.waitForCondition(new TestCondition() {
+                @Override
+                public boolean conditionMet() {
+                    return store2.get("a").endsWith("125") && store2.get("b").endsWith("34");
+                }
+            }, maxWaitMs, "wait for agg to be <a,125> and <b,34>");
```

Comment:
I had a similar thought but was not sure. But as you raise it, I think yes, this could happen.

It seems, the store could have 3 states: `125`, `1225` and `12125` -- and for `b` it would be `34`, `344`, and `3434` -- we could check this with an or-condition. Or did I miss any possible result?